graphium/data/datamodule.py

```diff
@@ -1959,8 +1961,11 @@ def get_data_hash(self):
         # so that the data cache does not need to be regenerated
         # when epoch_sampling_fraction has changed.
         for task in self.task_specific_args.keys():
-            if "epoch_sampling_fraction" in args[task].keys():
-                args[task].pop("epoch_sampling_fraction")
+            try:
+                if "epoch_sampling_fraction" in args[task].keys():
+                    args[task].pop("epoch_sampling_fraction")
+            except:
+                pass
```

Comment:
I don't even understand the point of having a `try` and an `if` there. `Dict.pop` works even if the key is not available.

But we need to make sure that `args[task]` is not used elsewhere, even outside the current function since dict are passed as pointers. We only want to remove `epoch_sampling` for the hash key. So I would suggest the following.

```suggestion
            args[task] = deepcopy(args[task])
            args[task].pop("epoch_sampling_fraction")
```

Reply:
Indeed, this is not an ideal fix. I wanted to investigate the issue a bit more.

We cannot use `args[task].pop("epoch_sampling_fraction")` because `args[task]` may be of class `DatasetProcessingParams` instead of `Dict`. In particular, `ADMETBenchmarkDataModule` makes use of `DatasetProcessingParams`.

This originates from changes in 4b82ba3bb049b7ac1376f93e9537bd1a5d3842e0. We were using a Dict in all configs I saw, although I see a comment `# To be replaced by a new class "DatasetParams"` everywhere it appears.

Will create issue and think about a fix.